components/lab-secret-store/backend/src/lab_secret_store/secret_store/json_db_secret_store.py

```diff
@@ -14,4 +17,48 @@ def get_secret(self, project_id: str, secret_id: str) -> Secret:
             collection_id="_lab_secret_store_secrets",
             key=secret_id,
         )
-        return Secret.parse_raw(secret_doc.json_value)
+        res = Secret.parse_raw(secret_doc.json_value)
+        res.secret_text = decrypt(res.secret_text)
+        return res
+
+    def get_secrets(self, project_id: str) -> List[SecretMetadata]:
+        secret_docs = self.json_db.list_json_documents(
+            project_id=project_id,
+            collection_id="_lab_secret_store_secrets",
+        )
+        return [
+            SecretMetadata.parse_raw(secret_doc.json_value)
+            for secret_doc in secret_docs
+        ]
+
+    def create_secret(
+        self, project_id: str, secret_id: str, value: Secret
+    ) -> SecretMetadata:
+        value.secret_text = encrypt(value.secret_text)
+        return SecretMetadata.parse_raw(
+            self.json_db.create_json_document(
+                project_id=project_id,
+                collection_id="_lab_secret_store_secrets",
+                key=secret_id,
+                json_document=value.json(),
+                upsert=True,
+            ).json_value
+        )
+
+    def update_secret(
+        self, project_id: str, secret_id: str, value: SecretUpdate
+    ) -> JsonDocument:
+        value.secret_text = encrypt(value.secret_text)
```

Comment:
Here we need to check if `value.secret_text` was actually set. It could be that only the metadata should be updated. In that case `value.secret_text` would be "". So a check like `if value.secret_text != ""` should work. We should have a test case for that!